### PR TITLE
Feature/update user validation

### PR DESCRIPTION
**Pull request by dhelmert**
### Reason for this PR
- All users could edit fields such as matriculationId
- Validation was untested

### Changes in this PR
- Add check for editable fields in User (if invoked by non-admin)
- Add tests for validation

### CODE REVIEW

File: product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Lecturer.scala [hunk 1]
```diff
@@ -19,9 +20,39 @@ case class Lecturer(username: String,
...
+  }
+  
+  /** 
+    * Compares the object against the user parameter to find out if fields, which should only be changed by users with elevated privileges, are different.
+    * Returns a list of SimpleErrors[[de.upb.cs.uc4.shared.client.SimpleError]]
+    * 
+    * @param user 
+    * @return Filled Sequence of [[de.upb.cs.uc4.shared.client.SimpleError]]
+    */
+  def checkEditableFields (user: Lecturer): Seq[SimpleError] = {
```
Comment (david-buderus) [inline product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Lecturer.scala:46]: There is no reason to overwrite this method

File: product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Student.scala [hunk 2]
```diff
@@ -21,6 +22,54 @@ case class Student(username: String,
...
+    * Returns a list of SimpleErrors[[de.upb.cs.uc4.shared.client.SimpleError]]
+    * 
+    * @param user 
+    * @return Filled Sequence of [[de.upb.cs.uc4.shared.client.SimpleError]]
+    */
+  def checkEditableFields (user: Student): Seq[SimpleError] = {
+
+    var errors = List[SimpleError]()
+   
+    errors ++= super.checkEditableFields(user)
```
Comment (david-buderus) [inline product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Student.scala:57]: You could combine these three lines into one for better style and a little bit more efficiency

File: product_code/lagom/user_service/impl/src/test/scala/de/upb/cs/uc4/user/impl/UserStateSpec.scala [hunk 3]
```diff
@@ -50,7 +54,7 @@ class UserStateSpec extends ScalaTestWithActorTestKit(s"""
       probe2.expectMessage(Some(student0))
     }
 
-    "not add an already existing user" in {
+    "not add an an already existing user" in {
```
Comment (david-buderus) [inline product_code/lagom/user_service/impl/src/test/scala/de/upb/cs/uc4/user/impl/UserStateSpec.scala:57]: TYPO

File: product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Student.scala [hunk 4]
```diff
@@ -21,6 +22,57 @@ case class Student(username: String,
...
+    * Returns a list of SimpleErrors[[de.upb.cs.uc4.shared.client.SimpleError]]
+    * 
+    * @param user 
+    * @return Filled Sequence of [[de.upb.cs.uc4.shared.client.SimpleError]]
+    */
+  override def checkEditableFields (user: User): Seq[SimpleError] = {
+    if(user.isInstanceOf[Student]){
+      throw new Exception("Tried to parse a non-Student as Student.")
+    }
+    val student = user.asInstanceOf[Student]
```
Comment (ldklenner) [inline product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Student.scala:56]: You check if the user object is a `Student`, but the exception shall be thrown when the object is not a `Student`.

**Pull request merged**